### PR TITLE
Ewmh urgent v1

### PR DESCRIPTION
**Pull request by Elv13**
Add better support for the "urgent" property for tags. It currently requires too much code for every taglist implementation.
This pull request adds the necessary code to better track urgent tags. It also adds some code to prevent noisy clients from setting themselves as urgent when they have the focus. While there may be some good use cases, I think it is usually undesired.

Finally, I also add awful.tag.swap, umm, because I can!

### CODE REVIEW

**Comment by blueyed** [comment]
@Elv13 
Please address my questions / comments and rebase it on current master.
I'd really like to have this for the "do not set current client as urgent" part.

**Comment by Elv13** [comment]
@blueyed Ok, I fixed the whitespaces (10 months late). I don't think moving emit to set_urgent is such a good idea.

As discussed with someone on IRC the other day, the same will have to be done for minimized as some buggy wine apps apparently auto minimize themselves.

**Comment by blueyed** [comment]
> the same will have to be done for minimized

Do you mean that "minimized" should get moved to ewmh/lua, too?

Apart from that this looks good to me.
But it's not based on master, is it? (because it cannot be merged automatically)

File: lib/awful/tag.lua.in [hunk 1]
```diff
@@ -630,7 +630,40 @@ capi.client.connect_signal("manage", function(c)
     c:connect_signal("property::screen", tag.withcurrent)
 end)
 
+-- Keep track of the number of urgent client
```
Comment (blueyed) [inline lib/awful/tag.lua.in]: s/client/clients/

**Comment by blueyed** [comment]
Thanks. It still looks good to me, apart from one typo and the file mode change:

> property.h 100644 → 100755

I'd like to see another review of it though.  Let's wait a few days, but if there's no feedback then let's merge it.

**Comment by blueyed** [comment]
btw: this would then need an update of the documentation, e.g. at http://awesome.naquadah.org/wiki/Signals.

Is the wiki the only place where signals are documented?

**Comment by psychon** [comment]
Looks fine to me

**Comment by blueyed** [comment]
Merged in https://github.com/awesomeWM/awesome/commit/f2aa8d39edf1427a635a0be3cbc96c7b2f54eb5b.

**Pull request merged**